saleor/graphql/checkout/mutations/checkout_create.py

```diff
@@ -308,8 +336,37 @@ def retrieve_billing_address(
     def clean_input(cls, info: ResolveInfo, instance: models.Checkout, data, **kwargs):
         user = info.context.user
         channel = data.pop("channel")
+
+        metadata_list: list[MetadataInput] = data.pop("metadata", None)
+        private_metadata_list: list[MetadataInput] = data.pop("private_metadata", None)
```

Comment:
The types of those fields are invalid. In case the `metadata` or `private_metadata` are not present in `data`, the `None` will be assigned, so the type of those fields is actually `None | list[MetadataInput]`.